pkg/authentication/basic/htpasswd.go

```diff
@@ -30,59 +33,114 @@ type sha1Pass string
 // NewHTPasswdValidator constructs an httpasswd based validator from the file
 // at the path given.
 func NewHTPasswdValidator(path string) (Validator, error) {
+	h := &htpasswdMap{users: make(map[string]interface{})}
+
+	err := h.loadHTPasswdFile(path)
+	if err != nil {
+		return nil, err
```

Comment:
Please wrap errors you are returning with additional context
```suggestion
		return nil, fmt.Errorf("could not load htpasswd file: %v", err)
```

Reply:
We already have the context in the `oauthproxy.go` file -> https://github.com/oauth2-proxy/oauth2-proxy/blob/master/oauthproxy.go#L114-L117.
We just need the value of the error retrieved from `loadHTPasswdFile` or `WatchFileForUpdates`.

Reply:
Changed the error message from `oauthproxy.go` and added the context for load and watch.